groups/memory_group_service.go

```diff
@@ -0,0 +1,201 @@
+package groups
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/topfreegames/pitaya/config"
+	"github.com/topfreegames/pitaya/constants"
+)
+
+var (
+	memoryGroupsMu sync.RWMutex
+	memoryGroups   map[string]*MemoryGroup
+	memoryOnce     sync.Once
+)
+
+// MemoryGroupService base in server memory solution
+type MemoryGroupService struct {
+}
+
+// MemoryGroup is the struct stored in each group key(which is the name of the group)
+type MemoryGroup struct {
+	Uids        []string
+	LastRefresh int64
+	TTL         int64
+}
+
+// NewMemoryGroupService returns a new group instance
+func NewMemoryGroupService(conf *config.Config) *MemoryGroupService {
+	memoryOnce.Do(func() {
+		memoryGroups = make(map[string]*MemoryGroup)
+		go func() {
```

Comment:
create a new function for that, it's a little ugly inside the constructor

Reply:
Okie Dokie!